cli/src/opts/cast.rs

```diff
@@ -437,6 +441,8 @@ Examples: 1ether, 10gwei, 0.01ether"#,
         #[clap(flatten)]
         // TODO: We only need RPC URL and Etherscan API key here.
         eth: EthereumOpts,
+        #[clap(subcommand)]
+        command: Option<EstimateSubcommands>,
```

Comment:
won't all arguments that `EstimateSubcommands` takes clash here?

like `forge estimate --sig .. create --sig ...`?

so we probably should disallow them here via `clap(conflicts_with = "..")`, but unsure about the ergonomics tbh.

Reply:
there's a [global](https://docs.rs/clap/2.33.0/clap/struct.Arg.html#method.global) flag that needs to specified for arguments to propogate into subcommands. Otherwise they are scoped to the command that they're defined at. 


At the match level, we reuse the variable names, but the `sig` and `args` variables are scoped to that `Create` match, so they don't conflict.